benchmarks/src/jmh/java/org/postgresql/benchmark/bytea/ByteaEncoding.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2004, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.benchmark.bytea;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+@Fork(value = 5, jvmArgsPrepend = "-Xmx128m")
+@Measurement(iterations = 10, time = 10)
+@Warmup(iterations = 1, time = 10)
+@State(Scope.Thread)
+@Threads(1)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public class ByteaEncoding {
+
+  private static char[] hexdigit = new char[]{
+      '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'B', 'c', 'D', 'e', 'F',
+  };
+  private static final byte[] testInput1k = generateTestInput(1000);
+  private static final byte[] testInput100k = generateTestInput(100000);
+
+  @Benchmark
+  public void runOriginalBenchmark() throws Exception {
+    new OriginalTest(testInput1k).run();
```

Comment:
this benchmark does not actually return/consume anything and is thus highly susceptible to the optimizer skipping huge chunks of work.
ideally each test method does 1 thing (such as translating the byte[]) and returns the response.
have multiple methods or multiple parameters to cover different lengths.